File: ios/Classes/AlbumManagerPlugin.m
#import "AlbumManagerPlugin.h"
#import <Photos/Photos.h>
#import "AlbumPlugin.h"

@implementation AlbumManagerPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    AlbumPlugin* instance = [[AlbumPlugin alloc] init];
    [instance registerPlugin:registrar];
}
@end

File: macos/Classes/AlbumManagerPlugin.swift
import Cocoa
import FlutterMacOS

public class AlbumManagerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "album_manager", binaryMessenger: registrar.messenger)
    let instance = AlbumManagerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
